pulsar/producer_partition.go

```diff
@@ -1041,6 +1041,12 @@ func (p *partitionProducer) internalFlushCurrentBatches() {
 }
 
 func (p *partitionProducer) internalFlush(fr *flushRequest) {
+	oldDataChan := p.dataChan
+	p.dataChan = make(chan *sendRequest, p.options.MaxPendingMessages)
+	for len(oldDataChan) != 0 {
+		pendingData := <-oldDataChan
+		p.internalSend(pendingData)
+	}
```

Comment:
```suggestion
	if len(p.dataChan) != 0 {
		oldDataChan := p.dataChan
		p.dataChan = make(chan *sendRequest, p.options.MaxPendingMessages)
		for len(oldDataChan) != 0 {
			pendingData := <-oldDataChan
			p.internalSend(pendingData)
		}
	}
```

Reply:
You r right. We should check length of dataChan first.